File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = max_freq = 0
        count = Counter()
        best = 0
        for right, ch in enumerate(s):
            count[ch] += 1
            max_freq = max(max_freq, count[ch])
            if(right - left + 1 - max_freq > k):
                count[s[left]] -= 1
                left += 1
            best = max(right - left + 1, best)
        return best

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int left = 0, maxFreq = 0, best = 0;
        array<int,26> count{};
        for(int right = 0; right<s.size(); right++){
            maxFreq = max(maxFreq, ++count[s[right] - 'A']);

            if(right - left + 1 - maxFreq > k){
                count[s[left] - 'A']--;
                left++;
            }
            best = max(best, right - left + 1);
        }
        return best;
    }
};